src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegmentStack.cs

```diff
@@ -20,6 +20,8 @@ public BufferSegmentStack(int size)
             _size = 0;
         }
 
+        public object Sync => _array;
```

Comment:
:+1: I was wondering if there was another self-contained object we could lock on. Add a comment here to explain what this is used for/by, and maybe that this struct shouldn't be shared.

Reply:
Done

Reply:
Had to revert as this breaks on resizes.